src/components/converter/Form.js

```diff
@@ -269,47 +310,70 @@ function FormSection() {
     }
 
     if (
-      amountToken &&
-      amountToken.gte(0) &&
-      amountToken.gt(selectedTokenBalance) &&
+      amountOther &&
+      amountOther.gte(0) &&
+      amountOther.gt(selectedTokenBalance) &&
       !selectedTokenBalance.eq(-1)
     ) {
       return 'Amount is greater than balance held.'
     }
 
     return null
   }, [
-    amountToken,
-    inputValueToken,
+    amountOther,
+    inputValueOther,
     balanceAnj,
     amountAnj,
     selectedTokenBalance,
   ])
 
   const disabled = Boolean(
-    !inputValueToken.trim() ||
+    !inputValueOther.trim() ||
       !inputValueAnj.trim() ||
       tokenBalanceError ||
       converterStatus.status !== CONVERTER_STATUSES.FORM ||
       !/[^@]+@[^@]+/.test(email) ||
       !acceptTerms
   )
 
+  const liquidityError = useMemo(() => {
+    // Reserves are still loading, so we cannot
+    // do any computation yet
+    if (loadingAnjReserve || !anjReserve || inputValueAnj.includes('Loading')) {
+      return false
+    }
+
+    return anjReserve.lt(amountAnj)
+      ? `There is not enough liquidity in the market at this time to purchase ${formatUnits(
+          amountAnj
+        )} ANJ. The maximum amount available for a purchase order at the current price is ${formatUnits(
+          anjReserve,
+          { truncateToDecimalPlace: 3 }
+        )} ANJ`
+      : ''
+  }, [amountAnj, anjReserve, loadingAnjReserve, inputValueAnj])
+
   const slippageWarning = useMemo(() => {
     const totalAmount = balanceAnj.add(amountAnj)
+
     const slippageWarning =
       totalAmount.gte(ANJ_MIN_REQUIRED) &&
-      calculateSlippageAmount(totalAmount).lt(ANJ_MIN_REQUIRED)
+      totalAmount
+        .sub(
+          totalAmount.mul(rateSlippage).div(bigNum(10).pow(UNISWAP_PRECISION))
+        )
```

Comment:
I think this calculation may be confused.

I think `amountAnj` has already taken into account the slippage (it is being calculated via the `executionRate`, right?).

In that case, we should show slippage warnings on `balanceAnj + amountAnj * 0.95 < minANJ` (where 0.95 is our internal "acceptable slippage rate").